meta_review/handler.py

```diff
@@ -0,0 +1,499 @@
+        to recalculate score.
+
+        :return: a bool indicates whether this is the first deploy.
+        """
+        # This query set is empty means it is the first deploy.
+        return not Participant.objects.all()
+
+    def _recalculate_score(self):
+        """
+        Recalculate score of each participant.
+
+        Used only on first deploy. Based on the weight factor derived,
+        recalculate score for all participants and comments.
+        """
+        self.logger.info('recalculate score of each participant')
+        # clear score of all comments and participants
+        for key, comment in self.comments.items():
+            comment.pos = 0
+            comment.weighted_pos = 0
+            comment.neg = 0
+            comment.weighted_neg = 0
+            comment.score = 0
+
+        for key, participant in self.participants.items():
+            participant.pos_in = 0
+            participant.weighted_pos_in = 0
+            participant.neg_in = 0
+            participant.weighted_neg_in = 0
+            participant.pos_out = 0
+            participant.neg_out = 0
+            participant.score = 0
+
+        # recalculate score and weight factor
+        self._update_score()
+        self._update_weight_factors()
+
+    def _update_score(self):
+        """
+        Update score of each participant.
+
+        Calculate and update score of each participant using
+        the following formula:
+
+        Define:
+
+        P1 = total points (weighted) of THUMBS_UP a person gets for all
+             reviews he did.
+        P2 = total number of THUMBS_UP a person gives to other
+             people for their reviews.
+        N1 = total points (weighted) of THUMBS_DOWN a person gets for all
+             reviews he did.
+        N2 = total number of THUMBS_DOWN a person gives to other people for
+             their reviews.
+
+        Then final score, denote by S, is as follows:
+
+        S = P1 - N1 + C1 * P2 + C2 * N2
+
+        where C1 = 0.05, C2 = 0.2. One will get at least 0.1 point for a
+        positive reaction they received, so we want C1 to be smaller than that.
+        C2 is larger because people are reluctant to give negative reactions.
+        In all, bonus points (P2 and N2) aim to encourage people to do
+        meta-reviews, but we don't want them to dominate.
+
+        Also update score of each review comment.
+        """
+        self.logger.info('update scores of all participants')
+
+        for key, participant in self.participants.items():
+            # parameters to be used in the formula
+            p1, p2, n1, n2 = 0, 0, 0, 0
+
+            # number of positive/negative reactions received
+            pos_cnt, neg_cnt = 0, 0
+
+            # get reactions received
+            reactions_in = participant.receive.all()
+            for reaction in reactions_in:
+                # skip old reactions since they were counted before
```

Comment:
create a method which is a generator of only new reactions not seen before.

That separates the logic regarding sync with the logic of processing items.